framework/Optimizers/GradientDescent.py

```diff
@@ -0,0 +1,854 @@
+    # __private
+    # additional methods
+    ## register adaptive sample identification criteria
+    self.registerIdentifier('purpose') # whether an opt, or which grad point
+
+  def handleInput(self, paramInput):
+    """
+      Read input specs
+      @ In, paramInput, InputData.ParameterInput, parameter specs interpreted
+      @ Out, None
+    """
+    RavenSampled.handleInput(self, paramInput)
+
+    # grad strategy
+    gradParentNode = paramInput.findFirst('gradient')
+    if len(gradParentNode.subparts) != 1:
+      self.raiseAnError('The <gradient> node requires exactly one gradient strategy! Choose from: ', gradKnownTypes())
+    gradNode = next(iter(gradParentNode.subparts))
+    gradType = gradNode.getName()
+    self._gradientInstance = gradReturnInstance(gradType, self)
+    self._gradientInstance.handleInput(gradNode)
+
+    # stepping strategy
+    stepNode = paramInput.findFirst('stepSize')
+    if len(stepNode.subparts) != 1:
+      self.raiseAnError('The <stepNode> node requires exactly one stepping strategy! Choose from: ', stepKnownTypes())
+    stepNode = next(iter(stepNode.subparts))
+    stepType = stepNode.getName()
+    self._stepInstance = stepReturnInstance(stepType, self)
+    self._stepInstance.handleInput(stepNode)
+
+    # acceptance strategy
+    acceptNode = paramInput.findFirst('acceptance')
+    if acceptNode:
+      if len(acceptNode.subparts) != 1:
+        self.raiseAnError('The <acceptance> node requires exactly one acceptance strategy! Choose from: ', acceptKnownTypes())
+      acceptNode = next(iter(acceptNode.subparts))
+      acceptType = acceptNode.getName()
+      self._acceptInstance = acceptReturnInstance(acceptType, self)
+      self._acceptInstance.handleInput(acceptNode)
+    else:
+      # default to strict mode acceptance
+      acceptNode = acceptReturnInstance('Strict', self)
+
+    # convergence options
+    convNode = paramInput.findFirst('convergence')
+    if convNode is not None:
+      for sub in convNode.subparts:
+        if sub.getName() == 'persistence':
+          self._requiredPersistence = sub.value
+        elif sub.getName() == 'terminateFollowers':
+          self._terminateFollowers = sub.value
+          self._followerProximity = sub.parameterValues.get('proximity', 1e-2)
+        elif sub.getName() == 'constraintExplorationLimit':
+          self._functionalConstraintExplorationLimit = sub.value
+        else:
+          self._convergenceCriteria[sub.name] = sub.value
+    if not self._convergenceCriteria:
+      self.raiseAWarning('No convergence criteria given; using defaults.')
+      self._convergenceCriteria['gradient'] = 1e-6
+    # same point is ALWAYS a criterion
+    self._convergenceCriteria['samePoint'] = 1e-16 # TODO user option?
+    # set persistence to 1 if not set
+    if self._requiredPersistence is None:
+      self.raiseADebug('No persistence given; setting to 1.')
+      self._requiredPersistence = 1
```

Comment:
We intend to add a default value option to the InputData in coming days; rather than delay this merge request, I recommend we make such changes then.

Reply:
I agree we will use InputData to do this in the future, the idea to move this to __init__ is to make this transition much easier, without going through all the source codes.

Reply:
I believe the current practice throughout the code has defaults set either in `__init__` or by the end of input reading. This has utility.

For example, I cannot do the check `if self._requiredPersistence is None` if the value is defaulted in `__init__`, hence I cannot provide this debug message if I'm using a default value. Since persistence is a large part of the convergence process, I feel like this is valuable feedback to provide to a user in debug mode.